cleanlab/internal/multiannotator_utils.py

```diff
@@ -89,11 +91,10 @@ def assert_valid_inputs_multiannotator(
 
         # Raise warning if no examples with 2 or more annotators agree
         # TODO: might shift this later in the code to avoid extra compute
-        if np.apply_along_axis(
-            lambda s: np.array_equal(np.unique(s[~np.isnan(s)]), s[~np.isnan(s)]),
-            axis=1,
-            arr=labels_multiannotator,
-        ).all():
+        has_agreement = np.zeros(labels_multiannotator.shape[0], dtype=bool)
+        for i in range(labels_multiannotator.shape[1]):
+            has_agreement |= (labels_multiannotator == i).sum(axis=1) > 1
```

Comment:
Thank you, that is a very good catch. I did not realize that it is very probable that there are more classes than annotators in a dataset. I have edited the code to ensure that all cases are correctly covered and I have also added one extra assert in the test code to catch this behavior.

Just in case I made a quick benchmark for that piece of code and it seems to still be much faster than the previous version and also the error I mentioned in the PR is now fixed and the function is warning about the issue correctly.
**Setup code** 
```python
import numpy as np

np.random.seed(0)
labels_multiannotator = np.random.randint(5, size=(1_000_000, 10))
```

```python
%%timeit
has_agreement = np.zeros(labels_multiannotator.shape[0], dtype=bool)
for i in np.unique(labels_multiannotator):
    has_agreement |= (labels_multiannotator == i).sum(axis=1) > 1

has_agreement.any()
# 266 ms ± 2.16 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
```python
%%timeit
np.apply_along_axis(
    lambda s: np.array_equal(np.unique(s[~np.isnan(s)]), s[~np.isnan(s)]),
    axis=1,
    arr=labels_multiannotator,
).all()
# 4.97 s ± 37.7 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```